Document limitations with the Google OIDC connector and transitive group memberships

espadolini opened the pull request:
Closes #10635, albeit unsatisfactorily.

espadolini committed:
Document limitations with Google OIDC

ptgott reviewed (commented):
It's not clear to me why this note should have the starting/ending point that it does. I think it would be clearer to break the "Configuring Google Workspace" H2 into H3-level sections. 

The first could be a "Configure Google group permissions" (or similar) section that rephrases the the six paragraphs after the "Configuring Google Workspace" heading into a set of instructions. 

Then we can change each bullet point into an H3-level heading so this organization is clearer. For example, the "Create a new project" bullet would instead be a "### Create a new project" H3 heading, then the first line in that section would be a sentence saying, "Next, [create a new project](https://console.cloud.google.com/projectselector2/apis/dashboard)."

What do you think? I'm also happy to implement this suggestions as a PR against your branch if that's easier for you.

espadolini commented:
The initial part of "Configuring Google Workspace" is currently also acting as an implicit "choose which mode of operation you want" step; it would be better if we made it more explicit - especially if we also tied it into the "upgrade" instructions that are currently kind of awkwardly placed at the end of "Create an OIDC connector".

Given my mediocre technical writing skills we'd probably be better off if you implemented those changes - feel free to just push to this PR rather than opening up a new one about it, my only goal was to have *something* documented about this weird behavior that we've discovered before more customers stumble upon it inadvertently, but from what I've seen, configuring the google workspace connector is still kind of a pain point so we should probably document it better.

zmb3 reviewed (approved):
Looks good to me as long as Paul's comments are addressed.

ptgott committed:
Reorganize the Google OIDC API instructions

- Organize the Google OIDC guide into steps. This required removing
  the header from the oidcauthentication.mdx partial.
- Use H3s for the "Configure Google Workspace" H2
- Add an explanation of the Workspace APIs

espadolini committed:
Fix inaccuracies

ptgott commented on docs/pages/enterprise/sso/google-workspace.mdx:
```diff
@@ -172,10 +172,11 @@ Google Admin Console, which relies on the Cloud Identity API.
-workspace level (following
-[these instructions](https://support.google.com/a/answer/9468710)) to prevent
-unexpected results when Teleport queries group memberships.
+When querying transitive group memberships we recommend blocking access to
```
```suggestion
When querying transitive group memberships, we recommend blocking access to
```

ptgott commented on docs/pages/enterprise/sso/google-workspace.mdx:
```diff
@@ -172,10 +172,11 @@ Google Admin Console, which relies on the Cloud Identity API.
-unexpected results when Teleport queries group memberships.
+When querying transitive group memberships we recommend blocking access to
+external groups at the workspace level (following [these
+instructions](https://support.google.com/a/answer/9468710)), as membership into
```
```suggestion
instructions](https://support.google.com/a/answer/9468710)), as membership in
```

ptgott reviewed (approved):
Left some minor suggestions, but this works for me!

espadolini committed:
PR suggestions

Co-authored-by: Paul Gottschling <paul.gottschling@goteleport.com>

espadolini committed:
Merge branch 'master' into espadolini/google-oidc-defeat
